as-vending/functions/output.go

```diff
@@ -64,15 +63,20 @@ func (vendingState *VendingState) HandleMqttDeviceReading(edgexcontext *appconte
 			"door:", vendingState.DoorClosed,
 		)
 
-		edgexcontext.LoggingClient.Debug("Processing reading from MQTT device service")
+		lc.Debug("Processing reading from MQTT device service")
 		for _, eventReading := range event.Readings {
-			switch eventReading.Name {
+			var simpleReading models.SimpleReading
+			var ok bool
+			if simpleReading, ok = eventReading.(models.SimpleReading); !ok {
+				return false, fmt.Errorf("event reading can not be casted to SimpleReading %v", reflect.TypeOf(eventReading))
+			}
+			switch simpleReading.DeviceName {
 			case "inferenceSkuDelta":
 				{
 					fmt.Println("Inference Started")
 					var skuDelta []deltaSKU
-					if err := json.Unmarshal([]byte(eventReading.Value), &skuDelta); err != nil {
-						edgexcontext.LoggingClient.Error("HandleMqttDeviceReading failed to unmarshal skuDelta message")
+					if err := json.Unmarshal([]byte(simpleReading.Value), &skuDelta); err != nil {
+						lc.Error("HandleMqttDeviceReading failed to unmarshal skuDelta message")
```

Comment:
it is probably beneficial for trouble shooting purpose to include what is the simpleReading.Value in the logging message. 
```suggestion
						lc.Error("HandleMqttDeviceReading failed to unmarshal skuDelta message %s: %v", simpleReading.Value, err)
```

Reply:
sorry in the suggestion it should be Errorf; not just Error